src/error/component_range.rs

```diff
@@ -77,7 +77,7 @@ impl serde::de::Expected for ComponentRange {
 #[cfg(feature = "serde")]
 impl ComponentRange {
     /// Obtain an "invalid value" error type for the deserializer.
-    pub(crate) fn to_invalid_serde_value<'a, D: serde::Deserializer<'a>>(self) -> D::Error {
+    pub fn to_invalid_serde_value<'a, D: serde::Deserializer<'a>>(self) -> D::Error {
```

Comment:
If we want users to be able to perform this conversion from outside of the crate (even inside a macro), we need to expose it.

As far as I can see our options are:
1. make them `pub`
2. reimplement inside of the macro each time

Unless I'm missing a better option. I'm happy to do (2) if you'd prefer.

Reply:
Yeah, it's unfortunate that Rust doesn't have the ability for workspace-public visibility, though I certainly understand the reasoning (it's more difficult and has more blockers than you'd expect). I suppose this could be exposed and be `#[doc(hidden)]` similar to a few other methods that are only `pub` for macros. In those situations they're `pub` for additional reasons, admittedly.

Reply:
I went with `#[doc(hidden)]`